react/features/facial-recognition/middleware.js

```diff
@@ -5,36 +5,53 @@ import {
     CONFERENCE_WILL_LEAVE,
     getCurrentConference
 } from '../base/conference';
+import { JitsiConferenceEvents } from '../base/lib-jitsi-meet';
 import { getParticipantCount } from '../base/participants';
 import { MiddlewareRegistry } from '../base/redux';
 import { TRACK_UPDATED, TRACK_ADDED, TRACK_REMOVED } from '../base/tracks';
-import { VIRTUAL_BACKGROUND_TRACK_CHANGED } from '../virtual-background/actionTypes';
 
-import { ADD_FACIAL_EXPRESSION } from './actionTypes';
+import { ADD_FACIAL_EXPRESSION, UPDATE_FACE_COORDINATES } from './actionTypes';
 import {
     addToFacialExpressionsBuffer,
-    changeTrack,
     loadWorker,
-    resetTrack,
     stopFacialRecognition,
     startFacialRecognition
 } from './actions';
+import { FACE_BOX_EVENT_TYPE } from './constants';
 import { sendFacialExpressionToParticipants, sendFacialExpressionToServer } from './functions';
 
 
 MiddlewareRegistry.register(({ dispatch, getState }) => next => action => {
-    const { enableFacialRecognition } = getState()['features/base/config'];
+    const { enableFacialRecognition, faceCoordinatesSharing } = getState()['features/base/config'];
+    const isEnabled = enableFacialRecognition || faceCoordinatesSharing?.enabled;
 
-    if (!enableFacialRecognition) {
-        return next(action);
-    }
     if (action.type === CONFERENCE_JOINED) {
-        dispatch(loadWorker());
+        if (isEnabled) {
```

Comment:
While looking into this I realized I haven't seen how the worker is stopped. Please test this with breakout rooms, since in that scenario we switch from one conference to another and may need to start / tear down the worker.

Reply:
good catch. tested across breakout rooms, we can keep reusing same worker instance, added a check in order to prevent reinitializing it.